src/RestApi/StoreApi/Controllers/CartShippingRates.php

```diff
@@ -0,0 +1,263 @@
+
+			if ( is_array( $valid_states ) && count( $valid_states ) > 0 ) {
+				$valid_state_values = array_map( 'wc_strtoupper', array_flip( array_map( 'wc_strtoupper', $valid_states ) ) );
+				$request['state']   = wc_strtoupper( $request['state'] );
+
+				if ( isset( $valid_state_values[ $request['state'] ] ) ) {
+					// With this part we consider state value to be valid as well,
+					// convert it to the state key for the valid_states check below.
+					$request['state'] = $valid_state_values[ $request['state'] ];
+				}
+
+				if ( ! in_array( $request['state'], $valid_state_values, true ) ) {
+					return new RestError(
+						'woocommerce_rest_cart_shipping_rates_invalid_state',
+						sprintf(
+							/* translators: 1: valid states */
+							__( 'Desintation state is not valid. Please enter one of the following: %s', 'woo-gutenberg-products-block' ),
+							implode( ', ', $valid_states )
+						),
+						[ 'status' => 400 ]
+					);
+				}
+			}
+		}
+
+		return $request;
+	}
+
+	/**
+	 * Cart item schema.
+	 *
+	 * @return array
+	 */
+	public function get_item_schema() {
+		return $this->schema->get_item_schema();
+	}
+
+	/**
+	 * Prepares a single item output for response.
+	 *
+	 * @param array            $package Shipping package complete with rates from WooCommerce.
+	 * @param \WP_REST_Request $request Request object.
+	 * @return \WP_REST_Response $response Response data.
+	 */
+	public function prepare_item_for_response( $package, $request ) {
+		return rest_ensure_response( $this->schema->get_item_response( $package ) );
+	}
+
+	/**
+	 * Get the query params available for this endpoint.
+	 *
+	 * @return array
+	 */
+	public function get_collection_params() {
+		$params                       = array();
+		$params['context']            = $this->get_context_param();
+		$params['context']['default'] = 'view';
+
+		$params['address_1'] = array(
+			'description'       => __( 'First line of the address being shipped to.', 'woo-gutenberg-products-block' ),
+			'type'              => 'string',
+			'default'           => '',
+			'sanitize_callback' => 'wc_clean',
+			'validate_callback' => 'rest_validate_request_arg',
+		);
+
+		$params['address_2'] = array(
+			'description'       => __( 'Second line of the address being shipped to.', 'woo-gutenberg-products-block' ),
+			'type'              => 'string',
+			'default'           => '',
+			'sanitize_callback' => 'wc_clean',
+			'validate_callback' => 'rest_validate_request_arg',
+		);
+
+		$params['city'] = array(
+			'description'       => __( 'City of the address being shipped to.', 'woo-gutenberg-products-block' ),
+			'type'              => 'string',
+			'default'           => '',
+			'sanitize_callback' => 'wc_clean',
+			'validate_callback' => 'rest_validate_request_arg',
+		);
+
+		$params['state'] = array(
+			'description'       => __( 'ISO code, or name, for the state, province, or district of the address being shipped to.', 'woo-gutenberg-products-block' ),
```

Comment:
It would simplify validation logic etc if we just restrict this to ISO code? We're only accepting ISO code for country so it'd be more consistent?

Reply:
I think I'd like to keep this for states. The way WC handles states its not always clear if it wants an ISO or a name - it varies between countries. In fact, most countries don't use ISO codes e.g. here we expect the county e.g. Cambridgeshire.